astropy/modeling/core.py

```diff
@@ -2943,29 +2939,14 @@ def rename(self, name):
     def isleaf(self):
         return False
 
-    def has_inverse(self):
-        return self._has_inverse
-
     @property
     def inverse(self):
```

Comment:
Is this handling user inverses properly? The` Model `class property for inverse does, but I assume this supersedes this. I don't see a test for `CompoundModel` user inverses in `test_compound.py`. But remember, I'm dim.

Reply:
It is handling user inverses, but the way it accomplishes that is a little... creative.  The `Model` metaclass pops the `CompoundModel` definition of `inverse` out of the class attributes and assigns it to `_inverse` instead.  So we're not actually overriding `Model`'s definition of `inverse`, even though it seems that way.

Reply:
OK, thanks, but still, I wonder if a test is needed if it doesn't exist. Not urgent of course.

Reply:
I think this might cover it:

https://github.com/astropy/astropy/blob/master/astropy/io/misc/asdf/tags/transform/tests/test_transform.py#L30

m1 is a `CompoundModel`, and we round trip that thing through an asdf file and compare the result.

Reply:
OK, good.